### PR TITLE
Ts functionality

### PR DESCRIPTION
**Pull request by plopdark**

### CODE REVIEW

**Comment by arsenja9** [comment]
Also then I will steal the code, and run around to try to fix bugs, as well as more carefully look at the layout + to understand the angular and add some comments.

**Comment by arsenja9** [comment]
So far, of such improvements, put repeating colours into one variable, and pull them down from there.

**Comment by arsenja9** [comment]
noticed a bug, when you buy an auto clicker, the money is not written off (maybe I had a joke, I don't know), but pay attention to it, as well as an improvement, make a limit on the auto clicker, so that it is not infinite (I think it's not difficult to do, it is supposed to be done on the back, but for the sake of interest you can try, you can limit the time just ).

File: src/app/pages/shop/shop.component.ts [hunk 1]
```diff
@@ -44,5 +68,71 @@ export class ShopComponent {
...
+    this.resetAsk = true;
+  }
+
+  public resetUpgrades() {
+    this.service.user.perClick = 1;
+    this.service.user.level = 1;
+    localStorage.setItem('user', JSON.stringify(this.service.user));
+    this.overlayClosed();
+  }
+
```
Comment (plopdark) [inline src/app/pages/shop/shop.component.ts:95]: Clicker login in component

**Comment by plopdark** [comment]
> noticed a bug, when you buy an auto clicker, the money is not written off (maybe I had a joke, I don't know), but pay attention to it, as well as an improvement, make a limit on the auto clicker, so that it is not infinite (I think it's not difficult to do, it is supposed to be done on the back, but for the sake of interest you can try, you can limit the time just ).

Fixed,check one of the comments

File: src/app/pages/account/account.component.scss [hunk 2]
```diff
@@ -37,3 +39,90 @@
     }
   }
 }
+
+@media (max-width: 850px) {
+  .account {
+    &__img {
```
Comment (Oskar2301) [inline src/app/pages/account/account.component.scss]: just .account__img

File: src/app/pages/account/account.component.scss [hunk 3]
```diff
@@ -37,3 +39,90 @@
...
+
+    &__info {
+      gap: 15px;
+    }
+  }
+}
+
+@media (max-width: 700px) {
+  .account {
+    &__top-position {
```
Comment (Oskar2301) [inline src/app/pages/account/account.component.scss:63]: the same

File: src/app/pages/account/account.component.scss [hunk 4]
```diff
@@ -37,3 +39,90 @@
...
+
+    &__top-position {
+      font-size: 18px;
+    }
+  }
+}
+
+@media (max-width: 700px) {
+  .account {
+    &__image {
```
Comment (Oskar2301) [inline src/app/pages/account/account.component.scss]: the same

File: src/app/pages/account/account.component.scss [hunk 5]
```diff
@@ -37,3 +39,90 @@
...
+      margin-right: 15px;
+    }
+  }
+}
+
+@media (max-width: 570px) {
+  .account {
+    &__img {
```
Comment (Oskar2301) [inline src/app/pages/account/account.component.scss]: +

File: src/app/pages/account/account.component.scss [hunk 6]
```diff
@@ -37,3 +39,90 @@
     }
   }
 }
+
```
Comment (Oskar2301) [inline src/app/pages/account/account.component.scss:42]: create mixin for media

File: src/app/pages/auth/auth.component.html [hunk 7]
```diff
@@ -0,0 +1,91 @@
+<div class="reg">
+  <form [formGroup]="loginForm" (ngSubmit)="onSubmit()" class="reg__form">
+    <h1 class="reg__title">Enter in your account</h1>
+    <div class="reg__inputs">
+      <input
```
Comment (Oskar2301) [inline src/app/pages/auth/auth.component.html:5]: move input in shared component, make this component multi functionality, and replace all input

File: src/app/pages/auth/auth.component.scss [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+    font-weight: 400;
+    color: red;
+  }
+}
+
+input.ng-invalid.ng-touched {
+  border: 0.9px solid red;
+}
+
+@media (max-width: 630px) {
```
Comment (Oskar2301) [inline src/app/pages/auth/auth.component.scss]: replace to mixin

File: src/app/pages/auth/auth.component.ts [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+        this.router.navigate([this.routerLinks.Main]);
+      }
+    }
+  }
+
+  public onFileSelected(event: any) {
+    if (event.target.files) {
+      const reader = new FileReader();
+      reader.readAsDataURL(event.target.files[0]);
+      reader.onload = (event: any) => {
```
Comment (Oskar2301) [inline src/app/pages/auth/auth.component.ts:67]: delete any

File: src/app/pages/main/main.component.scss [hunk 10]
```diff
@@ -99,3 +93,126 @@ a {
   color: #fff;
   text-decoration: none;
 }
+
+@media (max-width: 870px) {
```
Comment (Oskar2301) [inline src/app/pages/main/main.component.scss]: mixin

File: src/app/pages/shop/shop.component.ts [hunk 11]
```diff
@@ -20,18 +19,48 @@ import { ClickerService } from '../../shared/services/clicker.service';
...
+  public isAutoClickerActive = this.service.autoClickerActive;
 
   public readonly botImg = this.service.bot;
   public readonly clickImg = this.service.click;
 
-  constructor(
-    private readonly service: ClickerService,
-    private readonly main: MainService
-  ) {}
+  constructor(private readonly service: ClickerService) {
```
Comment (Oskar2301) [inline src/app/pages/shop/shop.component.ts]: move logic from constructor to ngOnInit and create separete method

File: src/app/pages/shop/shop.component.ts [hunk 12]
```diff
@@ -20,18 +19,48 @@ import { ClickerService } from '../../shared/services/clicker.service';
...
+      };
+    }
+
+    const clicking = localStorage.getItem('autoClicker');
+    if (clicking) {
+      this.isAutoClickerBought = JSON.parse(clicking);
+    }
+
+    const autoClickerActive = localStorage.getItem('autoClickerActive');
+    if (autoClickerActive === 'true') {
```
Comment (Oskar2301) [inline src/app/pages/shop/shop.component.ts:66]: ? why true in string?

File: src/app/shared/components/account-input/account-input.component.scss [hunk 13]
```diff
@@ -13,3 +13,77 @@
     font-size: 22px;
   }
 }
+
+@media (max-width: 940px) {
```
Comment (Oskar2301) [inline src/app/shared/components/account-input/account-input.component.scss]: mixin

File: src/app/shared/services/clicker.service.ts [hunk 14]
```diff
@@ -1,14 +1,94 @@
...
+  };
+
   public readonly icon = 'assets/images/default-person-icon.png';
   public readonly bot = 'assets/images/bot.png';
   public readonly click = 'assets/images/click.png';
-  public routerLinks = RouterLinkEnum;
 
-  constructor() {}
+  constructor() {
+    const userData = localStorage.getItem('user');
```
Comment (Oskar2301) [inline src/app/shared/services/clicker.service.ts:29]: the same, move in OnInit

File: src/app/shared/services/clicker.service.ts [hunk 15]
```diff
@@ -1,14 +1,94 @@
...
+      this.autoClickerActive = false;
+      localStorage.removeItem('autoClickerActive');
+    }, 60000);
+
+    this.autoClicker = setInterval(() => {
+      for (let i = 0; i < 1; i++) {
+        this.increaseClicks();
+      }
+    }, 1000 / 1);
+    localStorage.setItem('autoClickerActive', 'true');
```
Comment (Oskar2301) [inline src/app/shared/services/clicker.service.ts:84]: just true, not string

**Pull request merged**